### DISCUSSION TITLE
Building new entity in IReact engine

### DISCUSSION THREAD

**Discussion by RadekOlenski** [sebas77/Svelto.ECS#86] [Q&A] [answered]
Hi, I have a small question: do I understand correctly that if I build new entity in IReact engine (which is processed in submission phase), I have to wait until next SubmitEntities for that new entity to be submitted? It is ok to build entities in IReact engines or should I look for different place to do that?

**Comment by sebas77** [comment, marked as answer]
All the structural changes (creation, swap group, removal) happening inside a IReactOnSwap/Add/Remove(Ex) callbacks will be submitted in the same submission phase. Not more than 10 loops are allowed, so an exception will be thrown if a circular loop is found.
